contracts/RadiCards.sol

```diff
@@ -3,234 +3,504 @@ pragma solidity ^0.4.24;
-      _active
+    event LogClaim(
+            address indexed ephemeralAddress,
+            address indexed sender,
+            uint tokenId,
+            address receiver,
+            uint giftAmount,
+        bool daiDonation
     );
-    cardsIndex.push(_cardIndex);
-
-    emit CardAdded(_cardIndex);
-    return true;
-  }
-
-  function setTokenBaseURI(string _newBaseURI) external onlyIfWhitelisted(msg.sender) {
-    require(bytes(_newBaseURI).length != 0, "Base URI invalid");
 
-    tokenBaseURI = _newBaseURI;
-  }
+    event LogCancel(
+        address indexed ephemeralAddress,
+        address indexed sender,
+        uint tokenId
+    );
 
-  function setMinContribution(uint256 _minContribution) external onlyIfWhitelisted(msg.sender) {
-    require(_minContribution > 0, "Invalid minimum contribution");
 
-    minContribution = _minContribution;
-  }
+    event BenefactorAdded(
+        uint256 indexed _benefactorIndex
+    );
 
-  function setActive(uint256 _cardIndex, bool _active) external onlyIfWhitelisted(msg.sender) {
-    require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+    event CardAdded(
+        uint256 indexed _cardIndex
+    );
 
-    cards[_cardIndex].active = _active;
-  }
+    constructor () public ERC721Token("RadiCards", "RADI") {
+        addAddressToWhitelist(msg.sender);
+    }
+
+    function gift(address to, uint256 _benefactorIndex, uint256 _cardIndex, string _message, uint256 _donationAmount, uint256 _giftAmount, bool _claimableLink) payable public returns (bool) {
+        require(to != address(0), "Must be a valid address");
+        require(benefactors[_benefactorIndex].ethAddress != address(0), "Must specify existing benefactor");
+        require(bytes(cards[_cardIndex].tokenURI).length != 0, "Must specify existing card");
+        require(cards[_cardIndex].active, "Must be an active card");
+
+        Statuses _giftStatus;
+        address _sentToAddress;
+
+        if(_claimableLink){
+            require(_donationAmount + _giftAmount + EPHEMERAL_ADDRESS_FEE == msg.value, "Can only request to donate and gift the amount of ether sent + Ephemeral fee");
+            _giftStatus = Statuses.Deposited;
+            _sentToAddress = this;
+            ephemeralWalletCards[to] = tokenIdPointer;
+            to.transfer(EPHEMERAL_ADDRESS_FEE);
+        }
+
+        else {
+            require(_donationAmount + _giftAmount == msg.value,"Can only request to donate and gift the amount of ether sent");
+            _giftStatus = Statuses.Claimed;
+            _sentToAddress = to;
+        }
+
+        if (cards[_cardIndex].maxQnty > 0){ //the max quantity is set to zero to indicate no limit. Only need to check that can mint if limited
+            require(cards[_cardIndex].minted < cards[_cardIndex].maxQnty, "Can't exceed maximum quantity of card type");
+        }
+
+        if(cards[_cardIndex].minPrice > 0){ //if the card has a minimum price check that enough has been sent
```

Comment:
Also this may be better as a `modifier` as well

Reply:
@SoIidarity I think it should be `<` and not `<=` right as you can send the exact min amount?

Reply:
makes more sense to my personally the other way `msg.value >= minPrice`